DeepLinkSDK/DeepLink/DPLDeepLink.m

```diff
@@ -63,4 +65,60 @@ - (NSString *)description {
             [self.callbackURL description]];
 }
 
+
+#pragma mark - Equality
+
+- (BOOL)isEqual:(id)object {
+    if (self == object) {
+        return YES;
+    }
+    else if (![object isKindOfClass:[self class]]) {
+        return NO;
+    }
+    
+    return [self isEqualToDeepLink:object];
+}
+
+
+- (BOOL)isEqualToDeepLink:(DPLDeepLink *)deepLink {
+    if (!deepLink) {
+        return NO;
+    }
+    
+    return (((!self.URL && !deepLink.URL) ||
+             [self.URL isEqual:deepLink.URL]) &&
+            
+            ((!self.queryParameters && !deepLink.queryParameters) ||
+             [self.queryParameters isEqualToDictionary:deepLink.queryParameters]) &&
+            
+            ((!self.routeParameters && !deepLink.routeParameters) ||
+             [self.routeParameters isEqualToDictionary:deepLink.routeParameters]) &&
+            
+            ((!self.callbackURL && !deepLink.callbackURL) ||
+             [self.callbackURL isEqual:deepLink.callbackURL]));
+}
+
+
+- (NSUInteger)hash {
+    return [self.URL hash]
+    ^ [self.queryParameters hash]
+    ^ [self.routeParameters hash]
+    ^ [self.callbackURL hash];
+}
+
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone {
+    return [[DPLDeepLink alloc] initWithURL:self.URL];
+}
```

Comment:
Consider using `[self alloc]` in case this class is ever subclassed and then copied